namespace/data.go

```diff
@@ -0,0 +1,36 @@
+package namespace
+
+type PrefixedData struct {
+	namespaceLen uint8
```

Comment:
Is the type of namespace ID going to be abstracted to a type alias?

Reply:
Like this? https://github.com/lazyledger/nmt/blob/056112da486a306f9e88ac60f306025a9cebb9ed/namespace/id.go#L5

Or do you mean as an interface (which also could make sense)?

Reply:
In this case, for length, maybe `type NamespaceLen uint8`? That way you're not passing around raw `uint8`s without context.

Reply:
> That way you're not passing around raw uint8s without context.

That is true, but the param name also communicates some context. Also, unless the type alias doesn't also have Uint8 in its name, it won't communicate the min and may values here ([0, 255]). This reminds me, that I need to disallow `0` everywhere.

Reply:
Introduced a type alias in the namespace package: `type Size uint8`. So when using it, it will read: `namespace.Size`. You can still use `uint8` and `namespace.Size` more or less interchangeably...
regarding the 0 namespace size: this should just correspond to a regular (non-namespaced) merkle tree. Will add tests for that case too.